connectors/spark/src/main/scala/io/unitycatalog/spark/RetryableTemporaryCredentialsApi.java

```diff
@@ -0,0 +1,140 @@
+package io.unitycatalog.spark;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.unitycatalog.client.ApiException;
+import io.unitycatalog.client.api.TemporaryCredentialsApi;
+import io.unitycatalog.client.model.GenerateTemporaryPathCredential;
+import io.unitycatalog.client.model.GenerateTemporaryTableCredential;
+import io.unitycatalog.client.model.TemporaryCredentials;
+import io.unitycatalog.spark.utils.Clock;
+import java.time.Duration;
+import org.apache.hadoop.conf.Configuration;
+
+public class RetryableTemporaryCredentialsApi {
+  private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+  
+  private final TemporaryCredentialsApi delegate;
+  private final Clock clock;
+  private final int maxAttempts;
+  private final long initialDelayMs;
+  private final double multiplier;
+
+  @FunctionalInterface
+  private interface ApiCallSupplier {
+    TemporaryCredentials get() throws ApiException;
+  }
+
+  public RetryableTemporaryCredentialsApi(TemporaryCredentialsApi delegate, Configuration conf) {
+    this(delegate, conf, Clock.systemClock());
+  }
+
+  public RetryableTemporaryCredentialsApi(
+      TemporaryCredentialsApi delegate, Configuration conf, Clock clock) {
+    this.delegate = delegate;
+    this.clock = clock;
+    this.maxAttempts = conf.getInt(
```

Comment:
Nit:  Maybe we will need some kind of abstracted and common retry policy class in future ( for supporting other api, except the TemporaryCredentialsApi) , but currently I think it should be fine for us to have this customized retry code here.